app/models/news_article.rb

```diff
@@ -4,11 +4,33 @@ class NewsArticle < Node
            source_type: 'Section'
 
   alias_method :publisher, :section
-
   store_attribute :data, :release_date, Date
 
-  validates_presence_of :section, :parent
-  validates :parent, ancestry_depth: { minimum: 2 }
+  friendly_id :date_and_name, use: [:slugged, :scoped], routes: :default, scope: :section
+
+  validates_presence_of :section
 
   scope :published, -> { where state: 'published' }
+
+  def layout
+    'news_article'
+  end
+
+  # News Articles require specific redirects to a custom route that
+  # is not defined by node hierarchy. As such, this method is a convenience
+  # wrapper around the url helpers, to allow this model to override the default
+  # redirect action.
+  def full_path
```

Comment:
I'm a little lost on the distinction of the different approaches - yours would certainly work, but it doesn't really shift much more outside of the model. The model still has an implicit understanding of its path - all moving the call to the url_helper does it shift the call into the views. The actual knowledge of the path remains in the model.

Is there a Ruby idiom I'm missing that might help explain preferences to keep this kind of stuff outside of the model?